src/main/java/com/google/alpollo/SongDataBase.java

```diff
@@ -1,19 +1,28 @@
 package com.google.alpollo;
 
 import java.util.List;
+import java.util.ArrayList;
 
 /** Class that provides methods to work with the database. */
 public class SongDataBase {
   /** Number of the songs that will be shown to the user. */
   private static final int TOP_SIZE = 10;
 
-  /** Save request song to database. */
+  /** Save request song to database and increase the search counter. */
   public static void saveSongRequest(Song song) {
-    OfyService.ofy().save().entity(song).now();
+    SongCounter songCounter = OfyService.ofy().load().type(SongCounter.class).id(Song.id(song)).now();
+    if (songCounter == null) {
+      songCounter = new SongCounter(song);
+    }
+    songCounter.increaseSearchCounter();
+    OfyService.ofy().save().entity(songCounter).now();
   }
 
-  /** Returns the list of at most 10 songs. */
+  /** Returns the list of the most requested songs. */
   public static List<Song> topSongs() {
-    return OfyService.ofy().load().type(Song.class).limit(TOP_SIZE).list();
+    List<Song> songs = new ArrayList<>();
+    OfyService.ofy().load().type(SongCounter.class).order("-searchCounter").limit(TOP_SIZE).list()
+        .forEach(songCounter -> songs.add(songCounter.getSong()));
```

Comment:
Now `Song` class doesn't contain id field, which we need to know on the client side for retrieving `SongInfo`, can you, please, return `SongCounter`s instead: 
`return OfyService.ofy().load().type(SongCounter.class).order("-searchCounter").limit(TOP_SIZE).list();`